core/src/main/java/org/apache/calcite/sql/SqlDataTypeSpec.java

```diff
@@ -97,27 +104,34 @@ public SqlDataTypeSpec(
       final SqlTypeNameSpec typeNameSpec,
       TimeZone timeZone,
       SqlParserPos pos) {
-    this(typeNameSpec, timeZone, null, pos);
+    this(typeNameSpec, timeZone, /*nullable=*/ null, /*uppercase=*/ null,
+        /*caseSpecific=*/ null, pos);
   }
 
   /**
    * Creates a type specification representing a type, with time zone,
-   * nullability and base type name specified.
+   * nullability, uppercase status, caseSpecific status, and base type name specified.
    *
    * @param typeNameSpec The type name can be basic sql type, row type,
    *                     collections type and user defined type
    * @param timeZone     Specified time zone
    * @param nullable     The nullability
+   * @param uppercase    The uppercase status
```

Comment:
The existing param comments are admittedly already pretty weak, but I feel like we can add a little more information here :)